website/project/decorators.py

```diff
@@ -329,3 +329,20 @@ def wrapped(*args, **kwargs):
 
     # Return decorator
     return wrapper
+
+def must_have_permission_or_public_wiki(func):
```

Comment:
Perhaps rename to `must_have_write_permission_or_public_wiki` to indicate the necessary permission level.